src/Authentication/Traits/HasAccessTokens.php

```diff
@@ -165,4 +171,46 @@ public function setAccessToken(?AccessToken $accessToken): self
 
         return $this;
     }
+
+    /**
+     * Checks if the provided Access Token has expired.
+     *
+     * @return bool Returns true if Access Token has expired, false if not
+     */
+    public function hasAccessTokenExpired(?AccessToken $accessToken): bool
+    {
+        return $accessToken->expires !== null && $accessToken->expires->isBefore(Time::now());
+    }
+
+    /**
+     * Sets an expiration for Access Tokens by ID.
+     *
+     * @param int  $id        AccessTokens ID
+     * @param Time $expiresAt Expiration date
+     *
+     * @return bool Returns true if expiration date is set or updated.
+     */
+    public function setAccessTokenExpirationById(int $id, Time $expiresAt): bool
+    {
+        /** @var UserIdentityModel $identityModel */
+        $identityModel = model(UserIdentityModel::class);
+        $result        = $identityModel->setIdentityExpirationById($id, $this, $expiresAt, AccessTokens::ID_TYPE_ACCESS_TOKEN);
+
+        if ($result) {
+            // refresh currentAccessToken with updated data
+            $this->currentAccessToken = $identityModel->getAccessTokenById($id, $this);
+        }
+
+        return $result;
+    }
+
+    /**
+     * Checks if the current Hmac token can expire
+     *
+     * @return bool Returns true if AccessToken can expire.
+     */
+    public function canAccessTokenExpire(AccessToken $accessToken): bool
+    {
+        return isset($accessToken->expires);
```

Comment:
```suggestion
        return $accessToken->expires !== null;
```
Since the existence of the object is guaranteed, we can go with a strict check.